lib/routes/addRelation.js

```diff
@@ -31,7 +31,7 @@ addRelationRoute.register = function() {
         resourceConfig.handlers.find(request, callback);
       },
       function(ourResource, callback) {
-        theirResource = ourResource;
+        theirResource = JSON.parse(JSON.stringify(ourResource));
```

Comment:
That resource is the one that goes out the door to the client, so if it does have a circular include it'll crash further up the stack anyway.